packages/@stylexjs/babel-plugin/src/visitors/stylex-props.js

```diff
@@ -202,6 +202,113 @@ export default function transformStylexProps(
     // convert resolvedStyles to a string + ternary expressions
     // We no longer need the keys, so we can just use the values.
     const stringExpression = makeStringExpression(resolvedArgs);
+
+    // Check if this is used as a JSX spread attribute and optimize
+    // the output to avoid object creation and Babel helper overhead
+    if (path.parentPath.node.type === 'JSXSpreadAttribute') {
+      if (
+        t.isObjectExpression(stringExpression) &&
+        stringExpression.properties.length > 0 &&
+        stringExpression.properties.every(
+          (prop) =>
+            t.isObjectProperty(prop) &&
+            (t.isIdentifier(prop.key) || t.isStringLiteral(prop.key)) &&
+            !prop.computed,
+        )
+      ) {
+        const jsxElement = path.parentPath.parentPath;
+        if (jsxElement && t.isJSXOpeningElement(jsxElement.node)) {
+          const jsxOpeningElement = jsxElement.node;
+          const spreadAttributeIndex = jsxOpeningElement.attributes.findIndex(
+            (attr) => attr === path.parentPath.node,
+          );
+
+          // Get attribute names from `props` call
+          const stylexAttributeNames = new Set(
+            stringExpression.properties
+              .filter((prop) => t.isObjectProperty(prop))
+              .map((prop) => {
+                const objectProp = prop;
+                const key = objectProp.key;
+                if (t.isIdentifier(key)) {
+                  return key.name;
+                } else if (t.isStringLiteral(key)) {
+                  return key.value;
+                }
+                return null;
+              })
+              .filter(Boolean),
+          );
+
+          // Find existing attributes with same names
+          const attributesToRemove: number[] = [];
+          const attributesToSkip = new Set<string>();
+
+          jsxOpeningElement.attributes.forEach((attr, index) => {
+            if (t.isJSXAttribute(attr) && t.isJSXIdentifier(attr.name)) {
+              const attrName = attr.name.name;
+              if (stylexAttributeNames.has(attrName)) {
+                if (index < spreadAttributeIndex) {
+                  // Attribute before spread can be removed
```

Comment:
technically, it could have side effects:

```
{
  foo: alert('hi'),
  ...{ foo: 1 },
}
```
but probably okay to ignore?

Reply:
The alternative would be to just bail on duplicate arguments and not apply this optimization if you don't know what to safely do.